SatLink on Linux

gabrielribcesario opened the pull request:
- SatLink is now compatible with Linux. The main problem was the file path format used in some files ("\\" instead of "/");
- I also removed some deprecated functions (e.g. pandas.append);
- Refactored a small piece of code in the main_window.py file.

cfragoas commented:
I tested all the functions on Windows.
The Antenna Size List Calculation crashes upon completion. It might be an issue with my virtual machine (Process finished with exit code -1073740791 (0xC0000409)). I’ll test it again on a Windows laptop.
The program now works on macOS, but it only displays the window without the buttons. I believe all the calculations are functioning properly now.
It’s not working for me on Linux (qt.qpa.plugin: Could not load the Qt platform plugin “xcb” in ‘’’ even though it was found). I tested it on Ubuntu 2.24.
Could you add your package versions to the requirements?

gabrielribcesario commented:
I haven't had much time to look into it, but I've come up with a few updates:
- I think that the error code you we're getting during the antenna list calculation is the Windows equivalent of a core dump. It might've originated from a pandas.DataFrame.append() I've missed in the antenna_size.py file, line 183 to be exact (my bad). I've already fixed that piece of code;
- I've updated the requirements.txt list to better reflect the current pakages' versions I'm using in my virtual environment. I've also updated the README.md to reflect these changes;
- I've also added a one-liner for the Linux dependencies to the installation section of the README.md file, as well as some run of the mill debbuging info for some generic warnings/errors.
Just FYI I'm currently running these tests on Ubuntu 22.04 with WSL. I've already pushed these changes to my fork, lmk if I missed anything / something isn't working right.
